File: Java Basics/03HomeworkJavaLoopsMethodsClasses/JavaLoopsMethodsClasses/src/main/java/Product/Product.java
package Product;

import java.math.BigDecimal;

public class Product implements Comparable<Product> {

    private String name;

    private BigDecimal price;

    public Product(String name, BigDecimal price){
        this.setName(name);
        this.setPrice(price);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }



    @Override
    public String toString() {
        return String.format("%.2f %s",this.getPrice(),this.getName());
    }

    @Override
    public int compareTo(Product o) {
        return this.getPrice().compareTo(o.getPrice());
    }
}

File: Data Structures/10HomeworkDataStructureEfficiency/10HomeworkDataStructuresEfficiency/03CollectionOfProducts/Product.cs
namespace _03CollectionOfProducts
{
    using System;

    public class Product : IComparable<Product>
    {
        public Product(int id, string title, string supplier, double price)
        {
            this.Id = id;
            this.Title = title;
            this.Supplier = supplier;
            this.Price = price;
        }

        public int Id { get; private set; }

        public string Title { get; private set; }

        public string Supplier { get; private set; }

        public double Price { get; private set; }

        public int CompareTo(Product other)
        {
            return this.Id.CompareTo(other.Id);
        }

        public override string ToString()
        {
            return string.Format("{0} {1} {2} {3:F2}", this.Id, this.Title, this.Supplier, this.Price);
        }
    }
}
